internal/icinga2/client_api.go

```diff
@@ -91,10 +110,11 @@ func (client *Client) queryObjectsApi(
 		return nil, err
 	}
 
-	if res.StatusCode != http.StatusOK {
+	err = checkHTTPResponseStatusCode(res.StatusCode)
+	if err != nil {
 		_, _ = io.Copy(io.Discard, res.Body)
 		_ = res.Body.Close()
-		return nil, fmt.Errorf("unexpected HTTP status code %d", res.StatusCode)
+		return nil, err
```

Comment:
Doesn't this hide the actual status message from Icinga 2? I would either add the `resp.status` message to the error as well or simply derive an error from it, since it should contain enough information/context as to what went wrong.

Reply:
Nope, the error returned from `checkHTTPResponseStatusCode` contains the status code if it is seen as being invalid.

Reply:
Are you aware of `resp.Status` **<>** `resp.StatusCode`?